ownCloud/Settings/MediaUploadSettingsViewController.swift

```diff
@@ -18,21 +18,71 @@
 
 import UIKit
 import ownCloudSDK
+import ownCloudApp
 import ownCloudAppShared
 
 class MediaUploadSettingsViewController: StaticTableViewController {
 
+	private var proPhotoSettingsSection: StaticTableViewSection?
+
 	override func viewDidLoad() {
 		super.viewDidLoad()
 		navigationItem.title = "Media Upload".localized
 
+		if let userDefaults = OCAppIdentity.shared.userDefaults {
+			proPhotoSettingsSection = ProPhotoUploadSettingsSection(userDefaults: userDefaults)
+		}
+
 		if let userDefaults = OCAppIdentity.shared.userDefaults {
 			self.addSection(MediaExportSettingsSection(userDefaults: userDefaults))
+
 			if OCBookmarkManager.shared.bookmarks.count > 0 {
 				self.addSection(AutoUploadSettingsSection(userDefaults: userDefaults))
 				// TODO: Re-add this section when we re-gain an ability to run background NSURLSessions
 				//self.addSection(BackgroundUploadsSettingsSection(userDefaults: userDefaults))
 			}
+
+			reconsiderProPhotoSettingsSection()
 		}
+
+		NotificationCenter.default.addObserver(self, selector: #selector(bookmarksChanged), name: Notification.Name.OCBookmarkManagerListChanged, object: nil)
+	}
+
+	private func reconsiderProPhotoSettingsSection() {
+
+		guard let proSettingsSection = proPhotoSettingsSection else { return }
+
+		var addProPhotoSettings = false
+
+		if OCLicenseEnterpriseProvider.numberOfEnterpriseAccounts > 0 {
+			addProPhotoSettings = true
+		}
+
+		let featureAuthStatus = OCLicenseManager.shared.authorizationStatus(forFeature: .photoProFeatures, in: OCLicenseEnvironment())
+
+		switch featureAuthStatus {
+		case .unknown:
+			OCLicenseManager.shared.observeProducts(nil, features: [ .photoProFeatures ], in: OCLicenseEnvironment(), withOwner: self) { (licenseObserver, isInitial, _) in
+				// call code that adds and removes the section depending on authStatus and number of enterprise accounts
+				if isInitial {
+					OCLicenseManager.shared.stop(licenseObserver)
+				}
+				self.reconsiderProPhotoSettingsSection()
```

Comment:
When using `self` in this block, please make it `weak` to not cause a retain cycle (`OCLicenseManager` holds onto `block` - and `block` holds onto `MediaUploadSettingsViewController`).

Reply:
@felix-schwarz done